chef/cookbooks/neutron/recipes/common_agent.rb

```diff
@@ -172,21 +172,33 @@
     bound_if = (node[:crowbar_wall][:network][:nets][net[0]].last rescue nil)
     next unless bound_if
     name = "br-#{net[1]}"
-    execute "Neutron: create #{name}" do
-      command "ovs-vsctl add-br #{name}; ip link set #{name} up"
-      not_if "ovs-vsctl list-br |grep -q #{name}"
-    end
-    execute "Neutron: add #{bound_if} to #{name}" do
-      command "ovs-vsctl del-port #{name} #{bound_if} ; ovs-vsctl add-port #{name} #{bound_if}"
-      not_if "ovs-dpctl show system@#{name} | grep -q #{bound_if}"
+    ruby_block "Neutron: create #{name}" do
+      block do
+          Nic::OvsBridge.create(name)
```

Comment:
If you think switch to use Nic::OvsBridge here is to risky a change we could also stay here with calling ovs directly. But I think this makes the code (a little bit) cleaner.